### PR TITLE
Escape backslashes properly when converting to the DSL

### PR DESCRIPTION
**Pull request by DaveEwing**
Fix an issue where we weren't properly escaping backslashes in quote strings when converting to the DSL.

### CODE REVIEW

**Comment by DaveEwing** [comment]
@swift-ci please test

File: Sources/_StringProcessing/PrintAsPattern.swift [hunk 1]
```diff
@@ -578,7 +578,7 @@ extension PrettyPrinter {
 extension String {
   // TODO: Escaping?
   fileprivate var _quoted: String {
-    "\"\(self._replacing("\"", with: "\\\""))\""
+    "\"\(self._replacing("\\", with: "\\\\")._replacing("\"", with: "\\\""))\""
```
Comment (milseman) [inline Sources/_StringProcessing/PrintAsPattern.swift:581]:
```suggestion
    "\"\(self._replacing(#"\"#, with: #"\\"#)._replacing(#"""#, with: #"\""#))\""
```

File: Tests/RegexTests/RenderDSLTests.swift [hunk 2]
```diff
@@ -114,4 +114,12 @@ extension RenderDSLTests {
         """)
     }
   }
+  
+  func testQuoting() throws {
+    try testConversion(#"\\\"a\""#, #"""
+      Regex {
+        "\\\"a\""
```
Comment (milseman) [inline Tests/RegexTests/RenderDSLTests.swift:121]: Is this the right conversion? The input is raw, so IIUC it has 3 full backslashes, then a quote then a full backslash and another quote.

File: Sources/_StringProcessing/PrintAsPattern.swift (hunk 1 above)
Comment (milseman) [reply Sources/_StringProcessing/PrintAsPattern.swift:581]: Or we can emit it as a raw string if it has any backslashes or quotes inside it. We could do a quick check to count the number of hashes prior to a backslash and emit a raw string with one additional one.

File: Tests/RegexTests/RenderDSLTests.swift (hunk 2 above)
Comment (DaveEwing) [reply Tests/RegexTests/RenderDSLTests.swift:121]: I'll update the test. The `\`s in `\"` in the regex to convert is ignored when parsing, but not necessary.

**Comment by DaveEwing** [comment]
@swift-ci please test

**Review by Azoy — APPROVED** [review]

**Review by milseman — APPROVED** [review]

**Pull request merged**